config/routes.rb

```diff
@@ -1,9 +1,12 @@
 # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 Rails.application.routes.draw do
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :new]
   end
-
+  post '/users/:user_id/posts', to: 'posts#create', as: 'create_user_post'
+  post '/users/:user_id/posts/:id', to: 'comments#create', as: 'create_post_comment'
```

Comment:
- [ ] There should be a **pair** of routes in the _config/routes.rb_ file for creating comments. Kindly improve upon this aspect similar to the one shown below:

```ruby
   POST   /posts/:post_id/comments(.:format)     comments#create
   GET   /posts/:post_id/comments/new(.:format)     comments#new
```